test/statetest/statetest_loader.cpp

```diff
@@ -264,12 +264,24 @@ state::Transaction from_json<state::Transaction>(const json::json& j)
 {
     state::Transaction o;
     from_json_tx_common(j, o);
+    if (const auto chain_id_it = j.find("chainId"); chain_id_it != j.end())
+        o.chain_id = from_json<uint8_t>(*chain_id_it);
     o.data = from_json<bytes>(j.at("input"));
     o.gas_limit = from_json<int64_t>(j.at("gas"));
     o.value = from_json<intx::uint256>(j.at("value"));
 
     if (const auto ac_it = j.find("accessList"); ac_it != j.end())
+    {
         o.access_list = from_json<state::AccessList>(*ac_it);
+        if (o.kind == state::Transaction::Kind::legacy)  // Upgrade tx type if tx has "accessList"
+            o.kind = state::Transaction::Kind::eip2930;
+    }
+
+    if (const auto type_it = j.find("type"); type_it != j.end())
+    {
+        if (o.kind != state::Transaction::Kind(from_json<uint8_t>(*type_it)))
```

Comment:
```suggestion
        if (stdx::to_underlying(o.kind) != from_json<uint8_t>(*type_it))
```
You cannot cast to enum because it is potential undefined behavior.

Reply:
`std::to_underlying` is available since `C++23`. I will change to avoid undef behavior.